### PR TITLE
#170947688 Make a comment on accommodation

### PR DESCRIPTION
**Pull request by EugeneMunya**
#### What does this PR do?
- Add functionality for allowing a traveler to make a comment on a specific accommodation
#### Description of Task to be completed?
- User can make comment on a specific accommodation
#### How should this be manually tested?
- open your terminal and run this command `git clone https://github.com/andela/knightsbn   backend.git` to clone repository
- run `cd knights-bn-backend` to get inside the project
- run `git fetch && git checkout ft-accommodation-feed-back-170947688` 
- run `npm install` to install dependencies
- check if you have PostgreSQL database if you don't have it install it from here [PostgreSQL](https://www.2ndquadrant.com/en/blog/pginstaller-install-postgresql/)
- create database
- create `.env` file in your project root directory
- follow `env.example` file to create environment variables in `.env` file and 
   assign them with all your database configurations
- run `npm run dropTables` to clean your database
- run `npm run createTables` to create tables in your new database you have created.
- run `npm run dev-start` to start the server.
- open your postman or any other API tester you have
- you can log in with default credentials below
```
{
    "email": "alain.maxime@gmail.com",
    "password": "Niyonkuru@1"
}
```
- view all available accommodation facilities by sending a GET request to this URL `http://localhost:4000/api/v1/view/accommodations`  
- chose the one you want to comment on by grabbing its ID and append it at the end of this URL `http://localhost:4000/api/v1/accommodation/comment/{id}` and sent POST request to it along with the comment contained in the body as shown below.
```
{
"comment":" bad service"
}
```

- to view, all  accommodations and their comments sent GET request to this URL `http://localhost:4000/api/v1/view/accommodations`
- to view specific accommodation along with its comments send a GET request to this URL `http://localhost:4000/api/v1/view/accommodation/1` the number one `(1)` you see here is accommodation `ID`.  you can replace it with another accommodation `ID`  you want to view

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947688](https://www.pivotaltracker.com/story/show/170947688)
#### Make a comment

Screen Shot 2020-03-17 at 07 23 56

#### View all accommodation with their comments

Screen Shot 2020-03-17 at 07 26 50

#### View a specific accommodation and its comments

Screen Shot 2020-03-17 at 07 28 52

### CODE REVIEW

**Comment by Niyonsengaeric** [comment]
Screen Shot 2020-03-19 at 12 52 28

@EugeneMunya  on requesterId you should save user id instead of null 
@Niyonsengaeric is not requester id is requestId

**Comment by T2Wil** [comment]
@EugeneMunya , I am commenting on the accommodation and it is successfully applied to the specified accommodation but why is the `requestId` equals to `null` in the comments section of the accommodation?
image

**Comment by EugeneMunya** [comment]
> @EugeneMunya , I am commenting on the accommodation and it is successfully applied to the specified accommodation but why is the `requestId` equals to `null` in the comments section of the accommodation?
> image

Comment table that I am using is the one used by @maximme  to comment on a request

File: src/controllers/accommodation.js [hunk 1]
```diff
@@ -16,8 +18,11 @@ export default class accomodationFacility {
 
   static async getSingleAccommodation(req, res) {
```
Comment (T2Wil) [inline src/controllers/accommodation.js:19]: hey @EugeneMunya , 
Add
```
try{  }
catch (error) {
      return handleError(res, error);
    }
```
to catch this error:
image

image

Comment (EugeneMunya) [reply src/controllers/accommodation.js:19]: I am working on it right now

Comment (T2Wil) [reply src/controllers/accommodation.js:19]: @EugeneMunya , can you make this error more descriptive???

image

**Review by T2Wil — APPROVED** [review]

File: src/controllers/accommodation.js (hunk 1 above)
Comment (EugeneMunya) [reply src/controllers/accommodation.js:19]: I did you can check for now

File: src/tests/accommodation/accommodationFeedback.test.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      expect(accommodationFeedBackSpy).to.have.been.calledWith(req, res);
+      accommodationFeedBackSpy.restore();
+    });
+    it('it should return 200 when you commented ', (done) => {
+      chai
+        .request(app)
+        .post('/api/v1/accommodation/comment/1')
+        .send({ comment: 'Bad service' })
+        .end((err, res) => {
+          expect(res.statusCode).to.equal(201);
```
Comment (kagabof) [inline src/tests/accommodation/accommodationFeedback.test.js:47]: You should expect not expect the status only.

**Review by kagabof — CHANGES_REQUESTED** [review]

File: src/tests/accommodation/accommodationFeedback.test.js (hunk 2 above)
Comment (EugeneMunya) [reply src/tests/accommodation/accommodationFeedback.test.js:47]: cool let me work on it

Comment (EugeneMunya) [reply src/tests/accommodation/accommodationFeedback.test.js:47]: I have tested everything you can check

**Review by kagabof — APPROVED** [review]
Great job.

**Pull request merged**